nixos/modules/services/networking/newt.nix

```
@@ -0,0 +1,89 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.newt;
+in
+{
+  options = {
+    services.newt = {
+      enable = lib.mkEnableOption "Newt";
+
+      id = lib.mkOption {
+        type = lib.types.str;
+        default = null;
+        description = ''
+          Newt ID generated by Pangolin to identify the client, shown on site creation in the dashboard.
+        '';
+      };
+      endpoint = lib.mkOption {
+        type = lib.types.str;
+        default = null;
+        description = ''
+          The endpoint where both Gerbil and Pangolin reside in order to connect to the websocket. The url of your Pangolin dashboard.
+        '';
+      };
+      # provide path to file to keep secrets out of the nix store
+      secretFile = lib.mkOption {
+        type = lib.types.nullOr lib.types.path;
+        default = null;
+        description = ''
+          Path to the file containing a unique secret (not shared and kept private), used to authenticate the client ID with the websocket in order to receive commands. Specified by Pangolin on site creation.
+        '';
+      };
+      logLevel = lib.mkOption {
+        type = lib.types.enum [
+          "DEBUG"
+          "INFO"
+          "WARN"
+          "ERROR"
+          "FATAL"
+        ];
+        default = "INFO";
+        description = "The log level to use.";
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    assertions =
+      map
+        (opt: {
+          assertion = cfg.${opt} != null;
+          message = "services.newt.${opt} must be provided when Newt is enabled.";
+        })
+        [
+          "secretFile"
+          "id"
+          "endpoint"
+        ];
```

```suggestion
        (builtins.attrNames cfg);
```
For consistency with your changes to the Pangolin module.

---

good point